functions/src/index.test.ts

```diff
@@ -465,6 +465,80 @@ describe("token-service app", () => {
       expect(await getResourcesCount()).toEqual(1);
     });
 
+    it("creates a new resource with 'user' and 'context` access role type if user is authenticated (JWT)", async () => {
+      const settings = await createS3Settings();
+      const createQuery = {
+        type: settings.type,
+        tool: settings.tool,
+        name: "test.txt",
+        description: "desc"
+      };
+      const response = await supertest(webApiV1)
+        .post("/api/v1/resources")
+        .set("Authorization", `Bearer ${jwtToken}`)
+        .query({ env: "dev" })
+        .send(Object.assign({}, createQuery, {accessRuleType: ["user", "context"]}) as CreateQuery)
+        .expect(201);
+
+      const json = checkResponse(response);
+      const id = json.result.id;
+      expect(id).toEqual(expect.any(String));
+      expect(json.result).toEqual(expect.objectContaining({
+        id,
+        name: createQuery.name,
+        description: createQuery.description,
+        publicPath: `test-folder/${id}/`,
+        publicUrl: `https://test-bucket.s3.amazonaws.com/test-folder/${id}/`,
+        bucket: settings.bucket,
+        folder: settings.folder,
+        region: settings.region,
+        tool: settings.tool,
+        type: settings.type
+      }));
+      expect(json.result.accessRules.length).toEqual(2);
+      expect(json.result.accessRules[0]).toEqual({ type: "user", role: "owner", userId: user.userId, platformId: user.platformId } as UserAccessRule);
+      expect(json.result.accessRules[1]).toEqual({ type: "context", platformId: user.platformId, contextId: user.contextId } as ContextAccessRule);
+
+      expect(await getResourcesCount()).toEqual(1);
+    });
+
+    it("automatically adds 'user' access role type if user is authenticated and specified only 'context' access rule (JWT)", async () => {
```

Comment:
Is there a test for what happens if no accessRuleType is sent?
I'm not sure what it should do, but it seems like something we should define.

Reply:
The code already rejects such requests:
https://github.com/concord-consortium/token-service/blob/master/functions/src/resource-manager.ts#L99-L101

I've added a test that specifically checks that in my last commit:
https://github.com/concord-consortium/token-service/blob/2d300ecaefb54d0a65a5b80bf8d049a519060c3e/functions/src/index.test.ts#L567-L586

Reply:
looks good.